Source/Core/DolphinQt/GameList/GameFile.h

```diff
@@ -1,95 +1,93 @@
 class GameFile final
 {
 public:
-	GameFile(const QString& fileName);
-	GameFile(const std::string& fileName) : GameFile(QString::fromStdString(fileName)) {}
+	explicit GameFile(QString path);
+
+	bool IsValid()            const { return m_valid; }
+
+	// These will be properly initialized before we try to load the file.
+	QString GetPath()         const { return m_path; }
+	QString GetFileName()     const { return m_file_name; }
+	QString GetFolder()       const { return m_folder; }
+
+	// The rest will not.
+	QString GetUniqueID()     const { return m_unique_id; }
+	QString GetMakerID()      const { return m_maker_id; }
+	u16 GetRevision()         const { return m_revision; }
+	QString GetInternalName() const { return m_internal_name; }
+	QString GetCompany()      const { return m_company; }
+	u8 GetDiscNumber()        const { return m_disc_number; }
+	u64 GetRawSize()          const { return m_raw_size; }
+	QPixmap GetBanner()       const { return m_banner; }
+	QString GetIssues()       const { return m_issues; }
+	int GetRating()           const { return m_rating; }
 
-	bool IsValid() const { return m_valid; }
-	QString GetFileName() { return m_file_name; }
-	QString GetFolderName() { return m_folder_name; }
-	QString GetName(bool prefer_long, DiscIO::IVolume::ELanguage language) const;
-	QString GetName(bool prefer_long) const;
-	QString GetDescription(DiscIO::IVolume::ELanguage language) const;
-	QString GetDescription() const;
-	QString GetCompany() const { return m_company; }
-	u16 GetRevision() const { return m_revision; }
-	const QString GetUniqueID() const { return m_unique_id; }
-	const QString GetWiiFSPath() const;
-	DiscIO::IVolume::ECountry GetCountry() const { return m_country; }
 	DiscIO::IVolume::EPlatform GetPlatform() const { return m_platform; }
-	DiscIO::BlobType GetBlobType() const { m_blob_type; }
-	const QString GetIssues() const { return m_issues; }
-	int GetEmuState() const { return m_emu_state; }
-	bool IsCompressed() const
+	DiscIO::IVolume::ECountry GetCountry()   const { return m_country; }
+	DiscIO::BlobType GetBlobType()           const { return m_blob_type; }
+
+	DiscIO::IVolume::ELanguage GetLanguage() const;
+	QString GetShortName() const { return GetShortName(GetLanguage()); }
+	QString GetShortName(DiscIO::IVolume::ELanguage lang) const
 	{
-		return m_blob_type == DiscIO::BlobType::GCZ || m_blob_type == DiscIO::BlobType::CISO ||
-		       m_blob_type == DiscIO::BlobType::WBFS;
+		return m_short_names[lang];
 	}
-	u64 GetFileSize() const { return m_file_size; }
-	u64 GetVolumeSize() const { return m_volume_size; }
-	// 0 is the first disc, 1 is the second disc
-	u8 GetDiscNumber() const { return m_disc_number; }
-	const QPixmap GetBitmap() const
+
+	QString GetLongName() const { return GetLongName(GetLanguage()); }
+	QString GetLongName(DiscIO::IVolume::ELanguage lang) const
 	{
-		if (m_banner.isNull())
-			return Resources::GetPixmap(Resources::BANNER_MISSING);
+		return m_long_names[lang];
```

Comment:
GetLongName(ELanguage) returning an empty string if the language is missing might be fine, but it's not fine for GetLongName() since that's what the game list uses to get a name to display.